dependencies.gradle.kts

```diff
@@ -0,0 +1,96 @@
+import java.util.Properties
+
+val versionCodeDifferenceBetweenAppAndAutomotive = 50000
+val versionCodeDifferenceBetweenAppAndWear = 50000 + versionCodeDifferenceBetweenAppAndAutomotive
+
+val secretProperties = loadPropertiesFromFile(file("$rootDir/secret.properties"))
+val versionProperties = loadPropertiesFromFile(file("${rootDir}/version.properties"))
+
+fun loadPropertiesFromFile(file: File): Properties {
+    val properties = Properties()
+    if (file.exists()) {
+        file.inputStream().use { stream ->
+            properties.load(stream)
+        }
+    }
+    return properties
+}
+
+// Unfortunately we currently need to rely on a property to separate the version code between app & automotive builds.
+// This is mainly because the library modules use the version code information from BuildConfig directly instead
+// of receiving from the clients.
+//
+// On the bright side, the difference between the app & automotive modules is a static number, so even if there is
+// an issue and the wrong version code information is used, it'd be easy to identify it.
+//
+// The long term solution is to move the version information out of the `base.gradle` and into the specific
+// app & automotive modules and pass this information to the modules that require it.
+val isAutomotiveBuild = (project.findProperty("IS_AUTOMOTIVE_BUILD") as? String)?.toBoolean() ?: false
+val isWearBuild = (project.findProperty("IS_WEAR_BUILD") as? String)?.toBoolean() ?: false
+val getVersionCode = {
+    val appVersionCode = versionProperties.getProperty("versionCode", null).toInt()
+    if (isAutomotiveBuild) {
+        appVersionCode + versionCodeDifferenceBetweenAppAndAutomotive
+    } else if (isWearBuild) {
+        appVersionCode + versionCodeDifferenceBetweenAppAndWear
+    } else {
+        appVersionCode
+    }
+}
+val getVersionName = {
+    val versionName = versionProperties.getProperty("versionName", null)
+    if (isAutomotiveBuild) {
+        "${versionName}a"
+    } else if (isWearBuild) {
+        "${versionName}w"
+    } else {
+        versionName
+    }
+}
+
+project.apply {
+    extra.apply {
+        // Application
+        set("applicationId", "au.com.shiftyjelly.pocketcasts")
+
+        set("versionName", getVersionName())
+        set("versionCode", getVersionCode())
+
+        // Android
+        set("minSdkVersion", 23)
+        set("minSdkVersionWear", 26)
+        set("targetSdkVersion", 33)
+        set("compileSdkVersion", 33)
+        set("testInstrumentationRunner", "androidx.test.runner.AndroidJUnitRunner")
+
+        // App Signing
+        val storePath = secretProperties.getProperty("signingKeyStoreFile", null)
+        val storeFile = file(storePath)
```

Comment:
The old line seemed to work with absolute and relative paths. It had broken for my absolute path. 
```
storeFile = file("$rootDir/${secretProperties.getProperty("signingKeyStoreFile", null)}")
```
The `file` command documentation says the following so we don't require the `rootDir`. I have test this and it seems to work fine.

> Resolves a file path relative to the project directory of this project. This method converts the supplied path based on its type: 

https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:file(java.lang.Object)